PsfRuntime/CreateProcessHook.cpp

```diff
@@ -153,19 +157,68 @@ BOOL WINAPI CreateProcessFixup(
         processInformation = &pi;
     }
 
-    if (!CreateProcessImpl(
-        applicationName,
-        commandLine,
-        processAttributes,
-        threadAttributes,
-        inheritHandles,
-        creationFlags | CREATE_SUSPENDED,
-        environment,
-        currentDirectory,
-        startupInfo,
-        processInformation))
+    auto appConfig = PSFQueryCurrentAppLaunchConfig(true);
+    bool preventBreakAway = false;
+    auto preventBreakAwayPtr = appConfig->try_get("InPackageContext");
+    if (preventBreakAwayPtr)
+    {
+        preventBreakAway = preventBreakAwayPtr->as_boolean().get();
+    }
+
+    if (preventBreakAway)
     {
-        return FALSE;
+        STARTUPINFOEX startupInfoEx =

```

Comment:
StartupInfoEx and the call to CreateProcessImpl can be moved outside of the if block.

Make StartupInfoEx
if(Prevent Breakaway)
    Add breakaway
CreateProcessImpl

Reply:
you mean to use extended startupInfo even when prevent breakaway is not enabled ?

Reply:
Yes.  Then you can move the StartupInfoEx outside of the if block.